### PR TITLE
Add infresnel as a travel time computation plugin

### PR DESCRIPTION
**Pull request by liamtoney**
[_infresnel_](https://github.com/liamtoney/infresnel) is a Python package which computes the shortest diffracted paths over topography for arbitrary source and receiver locations and topography. The lengths of these paths can be used to estimate infrasound travel times over topography. This can be a pretty good estimate of the "true" travel times obtained by fancier fully 3D wavefield simulations. The benefit of using _infresnel_ is that it is (1) fast and (2) fully Python based. This makes it easier for folks to use.

This PR adds _infresnel_ as a travel time computation plugin, and includes a new example notebook [`example_local_infresnel.ipynb`](https://github.com/uafgeotools/rtm/blob/infresnel/example_local_infresnel.ipynb), showing how to use this option. This options includes a facility to store computed path length difference grids as NetCDF (`.nc`) files so that they can be loaded in rather than re-computed.

### CODE REVIEW

**Comment by liamtoney** [comment]
Pinging @darren-tpk as another reviewer

**Comment by darren-tpk** [comment]
Looks good to me! I was able to run the `example_local_infresnel.ipynb` notebook without problems, and reproduce some results that I had computed for Pavlof. 

Minor suggestion -- across the example notebooks, should we advocate for the normalization of `S`? That is, to add a line where `S.data = S.data / len(st_proc)`, then update `height=3` to `height=0.6` in `get_peak_coordinates()`. That would allow new users to tinker with the example notebook and test new cases with a different number of stations more easily.

**Comment by darren-tpk** [comment]
ok, I ran into a small issue today. The new utility in loading in pre-computed path lengths checks for the shape of the loaded `DataArray`. Specifically, the shape of `diff_path_lens_loaded` in here: https://github.com/uafgeotools/rtm/blob/30fb8f1272a407176be941f8201c4b50bf1a6708/rtm/travel_time.py#L365C13-L376C98

I was running `rtm` with `infresnel` with 5 stations in a loop, but ran into a time period where a partial network outage left me with 2 stations, causing the loaded path lengths to fail the shape check. 

A potential fix would be to select corresponding layers in the path lengths `DataArray` using trace IDs, prior to the shape check: 
`diff_path_lens_loaded = diff_path_lens_loaded.sel(station=[tr.id for tr in st])` 
which would allow the code to handle missing stations. 
Thoughts? Trying to think if this added flexibility could be problematic...

**Comment by liamtoney** [comment]
> Minor suggestion -- across the example notebooks, should we advocate for the normalization of `S`? That is, to add a line where `S.data = S.data / len(st_proc)`, then update `height=3` to `height=0.6` in `get_peak_coordinates()`. That would allow new users to tinker with the example notebook and test new cases with a different number of stations more easily.

We could do this, but we should make separate issue for it.

**Comment by liamtoney** [comment]
> I was running `rtm` with `infresnel` with 5 stations in a loop, but ran into a time period where a partial network outage left me with 2 stations, causing the loaded path lengths to fail the shape check.
> 
> A potential fix would be to select corresponding layers in the path lengths `DataArray` using trace IDs, prior to the shape check: `diff_path_lens_loaded = diff_path_lens_loaded.sel(station=[tr.id for tr in st])` which would allow the code to handle missing stations. Thoughts? Trying to think if this added flexibility could be problematic...

I think your proposal makes sense. I think we'd just want to make sure that the provided `st` contains **ONLY** a subset of the `station` dimension of the `diff_path_lens_loaded` array. In other words, we assume that the stored NetCDF file was computed for a full network, and that the only departure from that is when some of the full network's stations are missing.

I did some testing and I think your proposed implementation satisfies the above constraints, since it raises a `KeyError` if there's a station in the `st` that's not in the `station` dimension of `diff_path_lens_loaded`. Note that in addition to the shape check, we also check the exact values of the dimension, so we have some backup checks in place.

**Comment by liamtoney** [comment]
Okay @darren-tpk, I've made that changed and verified that it works w/ the _infresnel_ example. Can you test on your end please?

**Comment by davidfee5** [comment]
> Looks good to me! I was able to run the `example_local_infresnel.ipynb` notebook without problems, and reproduce some results that I had computed for Pavlof.
> 
> Minor suggestion -- across the example notebooks, should we advocate for the normalization of `S`? That is, to add a line where `S.data = S.data / len(st_proc)`, then update `height=3` to `height=0.6` in `get_peak_coordinates()`. That would allow new users to tinker with the example notebook and test new cases with a different number of stations more easily.

I agree with this! Perhaps a separate PR is needed. I have a question on filtering to bring up too.

**Comment by darren-tpk** [comment]
> Okay @darren-tpk, I've made that changed and verified that it works w/ the _infresnel_ example. Can you test on your end please?

Just did, works great and with the subset of stations too!

**Comment by liamtoney** [comment]
Excellent! Thank you @darren-tpk for that suggestion, that's a good catch. @davidfee5 I'll defer to you whether you'd like to also review this PR before it is merged.

File: example_local_infresnel.ipynb
Comment (davidfee5) [inline example_local_infresnel.ipynb:524]: It might be nice to note what a values of -2 means. I had to dig into the `infresnel` repo and see that -2 is listed as all cpus in `test_grid.py` only (I think).

**Review by davidfee5 — APPROVED** [review]
All looks good to me! I tried it on a separate example (Stromboli) and it worked well. Agreed on the other changes made here.

**Comment by liamtoney** [comment]
Good idea to list that in the docs. Great — I'll merge! Hooray

**Pull request merged**
